code/lib/core-server/src/build-dev.ts

```diff
@@ -102,6 +105,16 @@ export async function buildDevStandalone(
     getManagerBuilder(),
   ]);
 
+  if (builderName.includes('builder-vite')) {
+    const mainJsPath = serverResolve(resolve(options.configDir || '.storybook', 'main')) as string;
+    if (/\.c[jt]s$/.test(mainJsPath)) {
```

Comment:
This check isn't quite enough. You'd also need to check if they have a `main.js` that's CommonJS. You could check if it's CommonJS by seeing there's no `"type": "module"` in their `package.json` or if the file has a `require('@storybook/vite')` call in it (I'm probably not remembering the package name correctly)

Reply:
it's a bit tricky, because you could easily have a vaild config file that doesn't import anything.
However it always needs a default export.

What if we instead showed the notice if any of the following were true:

1. has `cjs` or `cts` file extension
2. has `module.exports` somewhere in the file
3. has `exports.default` somewhere in the file

?

Reply:
yeah, you're right. how about if we use the same regex as vite. it should be pretty battle tested

https://github.com/vitejs/vite/blob/91a18c2f7da796ff8217417a4bf189ddda719895/packages/vite/src/node/ssr/ssrExternal.ts#L87

Reply:
Sounds like a good idea.